### PR TITLE
Enhance/meta 4120 -Ability to move queries/folders across the Collection

### PR DESCRIPTION
**Pull request by sumandas0**
## Ability to move queries/folders across the Collection
In Atlan insights tab, we can move queries across folder. But we don't move across collection need to find out what are the constraints.

### CODE REVIEW

**Review by nixon-atlan — COMMENTED** [review]
Hope you have rebased your ranger-policy feature branch from master and then generated the ranger plugin jar ?

**Comment by sumandas0** [comment]
> Hope you have rebased your ranger-policy feature branch from master and then generated the ranger plugin jar ?

Merged the changes, also just once verify in beta if previous change is working properly @nixon-atlan

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/sql/QueryCollectionPreProcessor.java [hunk 1]
```diff
@@ -73,9 +73,7 @@ private void processCreate(AtlasStruct entity) {
     }
 
     private void processUpdate(AtlasStruct entity, AtlasVertex vertex) {
-        String vertexQnName = vertex.getProperty(QUALIFIED_NAME, String.class);
```
Comment (nikhilbonte21) [inline repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/sql/QueryCollectionPreProcessor.java:76]: I did not get reason to remove this logic

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/sql/QueryFolderPreProcessor.java [hunk 2]
```diff
@@ -80,12 +91,191 @@ private void processCreate(AtlasStruct entity) throws AtlasBaseException {
...
+        entity = entityRetriever.mapSystemAttributes(entityVertex, entity);
+        entity.setAttributes(updatedAttributes);
+        requestContext.cacheDifferentialEntity(new AtlasEntity(entity));
+
+        AtlasEntityType entityType = typeRegistry.getEntityTypeByName(entity.getTypeName());
+
+        //Add the min info attributes to entity header to be sent as part of notification
+        if(entityType != null) {
+            AtlasEntity finalEntity = entity;
+            entityType.getMinInfoAttributes().values().stream().filter(attribute -> !updatedAttributes.containsKey(attribute.getName())).forEach(attribute -> {
```
Comment (nikhilbonte21) [inline repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/sql/QueryFolderPreProcessor.java:261]: can we use entityType.getMinInfoAttributes().keySet() to filter attributes here?

**Review by nikhilbonte21 — CHANGES_REQUESTED** [review]

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/sql/QueryFolderPreProcessor.java (hunk 2 above)
Comment (sumandas0) [reply repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/sql/QueryFolderPreProcessor.java:277]: Kept this to use this function `entityRetriever.getVertexAttribute(entityVertex, attribute);` which is more sophisticated to fetch attributes. That need whole `attribute` there.

File: repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/sql/QueryCollectionPreProcessor.java (hunk 1 above)
Comment (sumandas0) [reply repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/sql/QueryCollectionPreProcessor.java]: Yeah, shouldn't be. Reverted change

**Review by nikhilbonte21 — APPROVED** [review]

**Pull request merged**